Content.Shared/Pinpointer/SharedPinpointerSystem.cs

```diff
@@ -27,13 +34,42 @@ private void OnAfterInteract(EntityUid uid, PinpointerComponent component, After
         if (!args.CanReach || args.Target is not { } target)
             return;
 
-        if (!component.CanRetarget || component.IsActive)
-            return;
-
         // TODO add doafter once the freeze is lifted
         args.Handled = true;
-        component.Target = args.Target;
-        _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(args.User):player} set target of {ToPrettyString(uid):pinpointer} to {ToPrettyString(component.Target.Value):target}");
+
+        if (component.CanRetarget && !component.IsActive || _tagSystem.HasTag(args.Target.Value, "PinpointerScannable"))
+        {
+            component.Target = args.Target;
+        }
+
+        else if (TryComp<NukeCodePaperComponent>(args.Target, out var nukeCodePaperComponent))
+        {
+            if (nukeCodePaperComponent.Nuke == null)
+                return;
+
+            component.Target = nukeCodePaperComponent.Nuke;
```

Comment:
Pinpointer shouldn't be getting nukes hardcoded in like this, this is why pinpointer was made generic.

Reply:
The NukeCodePaperComponent is what stores the EntityID of the nuke when the code is written on it.  The pinpointer only retrieves this value upon interacting with said piece of paper, it's not hardcoded on the pinpointer itself.